lib/backend/k8s/conversion.go

```diff
@@ -224,18 +224,16 @@ func (c converter) podToWorkloadEndpoint(pod *kapiv1.Pod) (*model.KVPair, error)
 	profile := fmt.Sprintf("ns.projectcalico.org/%s", pod.ObjectMeta.Namespace)
 	workload := fmt.Sprintf("%s.%s", pod.ObjectMeta.Namespace, pod.ObjectMeta.Name)
 
-	// If the pod isn't ready, we can't parse it.
-	if !c.isCalicoPod(pod) {
-		return nil, fmt.Errorf("Pod is not ready / valid. pod=%s", pod.Name)
-	}
-
 	// Parse the Pod's IP address.
-	_, ipNet, err := cnet.ParseCIDR(fmt.Sprintf("%s/32", pod.Status.PodIP))
-	if err != nil {
-		log.WithFields(log.Fields{"ip": pod.Status.PodIP, "pod": pod.Name}).WithError(err).Error("Failed to parse pod IP")
-		return nil, err
+	ipNets := []cnet.IPNet{}
+	if c.hasIPAddress(pod) {
```

Comment:
or does this method only get called when isCalicoPod() is true?   In which case we should beef up (or add) a method doc block.